bundles/AdminBundle/Resources/public/js/pimcore/document/editables/areablock.js

```diff
@@ -851,19 +851,20 @@ pimcore.document.editables.areablock = Class.create(pimcore.document.area_abstra
 
     removeBlock: function (element) {
         let container = Ext.get(element);
+
+        let dialogBoxDiv = container.query('.pimcore_block_dialog[data-name="' + this.name + '"]')[0];
+        if (dialogBoxDiv) {
+            let dialogBoxId = dialogBoxDiv.dataset.dialogId;
+            this.removeEditableDialogbox(dialogBoxId);
```

Comment:
@dvesh3 JFI: I've moved that code block to the beginning, because actually `container.remove();` removed the `.pimcore_block_dialog` container as well, because it's a child. Your code worked for whatever reason, however I thought it might be a bit confusing 😉